### PR TITLE
Tokenize mod

### PR DESCRIPTION
**Pull request by zancas**

### CODE REVIEW

File: zcashrpc-typegen/src/tokenize.rs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            object(name, obj, acc).map(|x| (x.0, x.1, false))
+        }
+        otherwise => Err(error::QuizfaceAnnotationError {
+            kind: error::InvalidAnnotationKind::from(otherwise),
+            location: name.to_string(),
+        })?,
+    }
+}
+
+pub(crate) fn terminal(
```
Comment (AloeareV) [inline zcashrpc-typegen/src/tokenize.rs]: Do these need to be pub at all?

File: zcashrpc-typegen/src/tokenize.rs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    );
+    quote!(
+        #[derive(Debug, serde::Deserialize, serde::Serialize)]
+        pub enum #name_tokens {
+            #(#variant_idents_renames #variant_idents,)*
+        }
+    )
+}
+
+pub(crate) fn value(
```
Comment (AloeareV) [inline zcashrpc-typegen/src/tokenize.rs:46]: I think this is the only function that needs to be `pub(crate)`, I think the others can all be private. Other than that, this looks great!

**Review by AloeareV — COMMENTED** [review]
Like the changes!

File: zcashrpc-typegen/src/tokenize.rs (hunk 1 above)
Comment (zancas) [reply zcashrpc-typegen/src/tokenize.rs]: Take em out..   see what happens.

File: zcashrpc-typegen/src/tokenize.rs (hunk 2 above)
Comment (zancas) [reply zcashrpc-typegen/src/tokenize.rs:46]: I will try this.

File: zcashrpc-typegen/src/tokenize.rs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                    .map(crate::capitalize_first_char)
+                    .collect::<String>(),
+            ))
+            .into()
+        })
+        .collect::<Vec<TokenStream>>();
+    let variant_idents_renames = variants
+        .map(|x| format!("#[serde(rename = \"{}\")]", x).parse().unwrap())
+        .collect::<Vec<TokenStream>>();
+    #[rustfmt::skip]
```
Comment (AloeareV) [inline zcashrpc-typegen/src/tokenize.rs]: Hmm.. Is this really necessary?

Comment (AloeareV) [reply zcashrpc-typegen/src/tokenize.rs]: I think the default formatting of 
```
      let name_tokens = crate::callsite_ident(                                    
          &(if called_by_alias {                                                  
              format!("{}Response", name)                                         
          } else {                                                                
              name.to_string()                                                    
          }),                                                                     
      );
```
Is fine

Comment (zancas) [reply zcashrpc-typegen/src/tokenize.rs]: That's not the default.   That's the custom formatting that is enabled by the skip.

**Pull request merged**